src/core/cloudxr/python/runtime.py

```
@@ -60,6 +52,40 @@ def _sdk_path() -> str | None:
+    the process exits early.
+    """
+    loop = asyncio.get_running_loop()
+    deadline = loop.time() + timeout_sec
+
+    lock_file = os.path.join(openxr_run_dir(), "runtime_started")
+
+    while loop.time() < deadline:
+        if not process.is_alive():
+            return False
+
+        if os.path.isfile(lock_file):
```

Ruff (0.15.4)

[warning] Async functions should not use os.path methods, use trio.Path or anyio.path

(ASYNC240)